src/codegen/encoding/scale_encoding.rs

```diff
@@ -578,4 +580,88 @@ impl AbiEncoding for ScaleEncoding {
     fn is_packed(&self) -> bool {
         self.packed_encoder
     }
+
+    /// TODO: This is used and tested for error data (Error and Panic) only.
+    fn const_encode(&self, args: &[Expression]) -> Option<Vec<u8>> {
+        let mut result = vec![];
+        for arg in args {
+            match arg {
+                Expression::AllocDynamicBytes {
+                    initializer: Some(data),
+                    ty: Type::String | Type::DynamicBytes,
+                    ..
+                } => result.extend_from_slice(&data.encode()),
+                Expression::AllocDynamicBytes {
+                    initializer: Some(data),
+                    ty: Type::Slice(inner),
+                    ..
+                } if matches!(**inner, Type::Bytes(1)) => result.extend_from_slice(data),
+                Expression::NumberLiteral {
+                    ty: Type::Bytes(4),
+                    value,
+                    ..
+                } => {
+                    let bytes = value.to_bytes_be().1;
+                    result.extend_from_slice(&bytes[..]);
```

Comment:
That's  not right. Say the values is `0x0000fffff`, then bytes will be two bytes and you will be missing the two leading bytes. This case really needs a test

Reply:
Hmm yeah good point, thanks. While it definitively works for this limited application on error codes. It's a bit a pain to get this right, because it's not natively in SCALE. And `num-bigint` doesn't document about trailing 0 bytes. I have to look into this again.